pkg/api/v1/kubernetes/clusters/clusters_test.go

```diff
@@ -186,67 +227,72 @@ func TestKubernetesClusterService_CreateOrUpdate(t *testing.T) {
+		{"np bad 4", args{ctx, projectID, clusterName, k_ok, np_bad_4, m_ok, h_ok, e_ok}, want, true, false},
+		{"np bad 5", args{ctx, projectID, clusterName, k_ok, np_bad_5, m_ok, h_ok, e_ok}, want, true, false},
+		{"np bad 6", args{ctx, projectID, clusterName, k_ok, np_bad_6, m_ok, h_ok, e_ok}, want, true, false},
+		{"np bad 7", args{ctx, projectID, clusterName, k_ok, np_bad_7, m_ok, h_ok, e_ok}, want, true, false},
+		{"np bad 8", args{ctx, projectID, clusterName, k_ok, np_bad_8, m_ok, h_ok, e_ok}, want, true, false},
+		{"np bad 9", args{ctx, projectID, clusterName, k_ok, np_bad_9, m_ok, h_ok, e_ok}, want, true, false},
+		{"np bad 10", args{ctx, projectID, clusterName, k_ok, np_bad_10, m_ok, h_ok, e_ok}, want, true, false},
+		{"np bad 11", args{ctx, projectID, clusterName, k_ok, np_bad_11, m_ok, h_ok, e_ok}, want, true, false},
+		{"np bad 12", args{ctx, projectID, clusterName, k_ok, np_bad_12, m_ok, h_ok, e_ok}, want, true, false},
+
+		{"maintenance bad 1", args{ctx, projectID, clusterName, k_ok, np_ok, m_bad_1, h_ok, e_ok}, want, true, false},
+		{"maintenance bad 2", args{ctx, projectID, clusterName, k_ok, np_ok, m_bad_2, h_ok, e_ok}, want, true, false},
+		{"maintenance bad 3", args{ctx, projectID, clusterName, k_ok, np_ok, m_bad_3, h_ok, e_ok}, want, true, false},
+
+		{"h bad 1", args{ctx, projectID, clusterName, k_ok, np_ok, m_ok, h_bad_1, e_ok}, want, true, false},
+		{"h bad 2", args{ctx, projectID, clusterName, k_ok, np_ok, m_ok, h_bad_2, e_ok}, want, true, false},
+
+		{"e bad 1", args{ctx, projectID, clusterName, k_ok, np_ok, m_ok, h_ok, e_bad_1}, want, true, false},
 	}
+
+	var process *wait.Handler
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			// test create
 			{
-				gotRes, err := s.Clusters.Create(tt.args.ctx, tt.args.projectID, tt.args.clusterName, tt.args.clusterConfig, tt.args.nodePools, tt.args.maintenance, tt.args.hibernation, tt.args.extensions)
+				gotRes, w, err := s.Clusters.CreateOrUpdate(tt.args.ctx, tt.args.projectID, tt.args.clusterName, tt.args.clusterConfig, tt.args.nodePools, tt.args.maintenance, tt.args.hibernation, tt.args.extensions)
 				if (err != nil) != tt.wantErr {
 					t.Errorf("KubernetesClusterService.Create() error = %v, wantErr %v", err, tt.wantErr)
 					return
 				}
 				if !reflect.DeepEqual(gotRes, tt.wantRes) && !tt.wantErr {
 					t.Errorf("KubernetesClusterService.Create() = %v, want %v", gotRes, tt.wantRes)
 				}
-			}
-			// test update
-			{
-				gotRes, err := s.Clusters.Update(tt.args.ctx, tt.args.projectID, tt.args.clusterName, tt.args.clusterConfig, tt.args.nodePools, tt.args.maintenance, tt.args.hibernation, tt.args.extensions)
-				if (err != nil) != tt.wantErr {
-					t.Errorf("KubernetesClusterService.Create() error = %v, wantErr %v", err, tt.wantErr)
-					return
-				}
-				if !reflect.DeepEqual(gotRes, tt.wantRes) && !tt.wantErr {
-					t.Errorf("KubernetesClusterService.Create() = %v, want %v", gotRes, tt.wantRes)
+
+				if tt.useWait {
```

Comment:
Maybe add a comment here (or further down) to clarify that `process.Wait` is only called once for the only reason that this simplifies testing? Because some people like using unit tests as documentation for how to use an API. Also probably a good idea to add a check that `tt.useWait` is only true once so that nobody accidentally messes the tests up.

Reply:
I don't see a reason why process.Wait cannot be used more than once

Reply:
> I don't see a reason why process.Wait cannot be used more than once

It's at the moment outside of the for loop (outside of the `for _, tt := range tests` block), so it only gets called once, and it only picks up the last assignment `process = w` out of all the sub-tests. So I think without a comment it's easy to introduce a new test with `tt.useWait` set to true, which would then not call `process.Wait` on the previous test where it's set to true.